### PR TITLE
[SPARK-11547] set fetchSize default value in JDBCRDD

### PR DESCRIPTION
**Pull request by ghost**
if I do not set any properties manual for jdbc, when I show the data in the table, it will throw an exception like below:
Exception in thread "main" java.lang.NullPointerException
    at java.util.Hashtable.putAll(Hashtable.java:522)
    at org.apache.spark.sql.DataFrameReader.jdbc(DataFrameReader.scala:205)
    at org.apache.spark.sql.DataFrameReader.jdbc(DataFrameReader.scala:130)

my test code:
//sqlContext : An existing SparkContext
    val df1 = sqlContext.read.jdbc("jdbc:mysql://IP:PORT/XXX?user=XXX&password=XXX","tablename",null)
    df1.show

This PR Makes it easier to operate, especially for a novice.
https://issues.apache.org/jira/browse/SPARK-11547

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala [hunk 1]
```diff
@@ -369,7 +369,7 @@ private[sql] class JDBCRDD(
     val sqlText = s"SELECT $columnList FROM $fqTable $myWhereClause"
     val stmt = conn.prepareStatement(sqlText,
         ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY)
-    val fetchSize = properties.getProperty("fetchsize", "0").toInt
+    val fetchSize = properties.getProperty("fetchsize", "10").toInt
```
Comment (srowen) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala]: It's not clear it's efficient to limit this to 10 by default. How does this relate to the NPE though? it's due to a null properties object.

Comment (yueqianzhu) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala]: Yes, my mistake. It's due to null properties object.
Is this better to provide a jdbc API like below:
def jdbc(url: String, table: String): DataFrame = {
    jdbc(url, table, new Properties())
  }
I think "properties" meaning is a non essential,so I always use null to test first time.

thanks.

Comment (srowen) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala]: I think you could either change this to throw a more explicit error with `require`, or make a new object if none is supplied.

Comment (yueqianzhu) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala]: It's not an error,I just provide a simplest API if no any properties supplied.

Comment (srowen) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala]: I'm not sure I understand. This change is unrelated to the problem you're discussing. Are you going to update or close them?
